packages/single-line/src/SingleLineEditor.tsx

```diff
@@ -84,25 +95,24 @@ export function SingleLineEditor(props: SingleLineEditorProps) {
                 setValue(e.target.value);
               }}
             />
-            {props.withCharValidation && (
-              <div className={styles.validationRow}>
-                <CharCounter value={value || ''} checkConstraint={checkConstraint} />
-                <CharValidation constraints={constraints} />
-              </div>
-            )}
-            {props.withCharValidation === false && (
-              <div className={styles.validationRow}>
-                <CharCounter value={value || ''} checkConstraint={() => true} />
-              </div>
+            {validationsAreVisable && (
+              <>
+                {withCharValidation && (
+                  <div className={styles.validationRow}>
+                    <CharCounter value={value || ''} checkConstraint={checkConstraint} />
+                    <CharValidation constraints={constraints} />
+                  </div>
+                )}
+                {withCharValidation === false && (
+                  <div className={styles.validationRow}>
+                    <CharCounter value={value || ''} checkConstraint={() => true} />
+                  </div>
+                )}
+              </>
```

Comment:
I think we can clean up these conditionals a bit to something like:

```tsx
// new const for checkConstraint above the return of the component
const handleCheckConstraint = withCharValidation ? checkConstraint : () => true

// updated return
{withCharInformation && (
    <div className={styles.validationRow}>
      <CharCounter value={value || ''} checkConstraint={handleCheckConstraint} />
      {withCharValidation && <CharValidation constraints={constraints} />}
    </div>
)}
```

Reply:
Nice, yeah this change definitely cleans the code up.

- [ ] clean up conditionals.